### PR TITLE
Parser for pymatgen structures into DFTK structures

### PR DESCRIPTION
**Pull request by mfherbst**
Parser for pymatgen `lattice` and `structure` structures as they are used by `abipy` and `pymatgen`.

### CODE REVIEW

**Comment by mfherbst** [comment]
@antoine-levitt Are you ok if I merge that?

File: src/bzmesh.jl [hunk 1]
```diff
@@ -90,3 +90,17 @@ function bzmesh_ir_wedge(kgrid_size, lattice, composition...; tol_symmetry=1e-5)
...
     kcoords, ksymops
 end
+
+@doc raw"""
+Selects a kgrid_size to ensure a minimal spacing (in inverse Bohrs) between kpoints.
+Default is ``2π * 0.04 \AA^{-1}``.
+"""
+function kgrid_size_from_minimal_spacing(lattice, spacing=2π * 0.04 / units.Ǎ)
+    for d in 1:3
+        @assert norm(lattice[:, d]) != 0
```
Comment (antoine-levitt) [inline src/bzmesh.jl]: You don't need that, the following line will fail anyway

File: src/common/constants.jl [hunk 2]
```diff
@@ -1,2 +1,8 @@
 """The default search location for Pseudopotential data files"""
 DFTK_DATADIR = joinpath(dirname(pathof(DFTK)), "..", "data")
+
+# Commonly used constants. The factors convert from the respective unit to AU
+module units
+    eV = 0.03674932248  # Hartree
+    Ǎ = 1.8897261254578281  # Bohr
+end
```
Comment (antoine-levitt) [inline src/common/constants.jl]: Very cool idea! Maybe add a few more? Kelvins, Rydbergs and we should be OK.

File: src/energy_ewald.jl [hunk 3]
```diff
@@ -90,7 +90,7 @@ function energy_ewald(lattice, recip_lattice, charges, positions; η=nothing)
         gsh += 1
     end
     # Amend sum_recip by proper scaling factors:
-    sum_recip = sum_recip * 4T(π) / det(lattice)
+    sum_recip = sum_recip * 4T(π) / abs(det(lattice))
```
Comment (antoine-levitt) [inline src/energy_ewald.jl]: Good catch on all these

File: src/bzmesh.jl (hunk 1 above)
Comment (mfherbst) [reply src/bzmesh.jl]: Yeah with the very helpful `SingularException` :-P . I think getting the assertion and then finding the line with the explanatory comment is more helpful. But I can remove it if you want.

File: src/common/constants.jl (hunk 2 above)
Comment (mfherbst) [reply src/common/constants.jl]: Good idea.

File: src/bzmesh.jl (hunk 1 above)
Comment (antoine-levitt) [reply src/bzmesh.jl]: Hehe, as usual, I tend towards less code = better. But I'm fine with either

**Pull request merged**